src/HTSPConnection.cpp

```diff
@@ -567,7 +567,11 @@ void CHTSPConnection::Register ( void )
 
 fail:
   if (!m_suspended)
+  {
+    /* Don't immediately reconnect (spare server CPU cycles)*/
+    Sleep(SLOW_RECONNECT_INTERVAL);
     Disconnect();
```

Comment:
@ksooo Side note: Don't we need to disconnect even if we are suspending? Otherwise the connection might look ok on wake, even if our credentials are wrong?